scarb/src/compiler/plugin/proc_macro/compilation.rs

```diff
@@ -61,6 +67,34 @@ impl SharedLibraryProvider for Package {
             .path_unchecked()
             .join(lib_name))
     }
+
+    fn prebuilt_lib_path(&self) -> Result<Utf8PathBuf> {
+        let target_triple = target!();
+
+        let prebuilt_name = format!(
+            "{name}_v{version}_{target}{suffix}",
+            name = self.id.name,
+            version = self.id.version,
+            target = target_triple,
+            suffix = DLL_SUFFIX
+        );
+
+        let prebuilt_path = self
+            .root()
+            .join("target")
+            .join("scarb")
+            .join("cairo-plugin")
+            .join(prebuilt_name);
+
+        prebuilt_path
+            .exists()
+            .then_some(prebuilt_path)
+            .ok_or_else(|| anyhow!("prebuilt library not found"))
```

Comment:
If the path is expected to not be defined in some cases, this should be an `Option` instead of `Result`. Let's handle the error where path missing is an actual error.